### DISCUSSION TITLE
Dynamic Session Cookies

### DISCUSSION THREAD

**Discussion by ghost** [projectdiscovery/nuclei#808] [Q&A] [answered]
Dear Team,

Just curious if there is a way to set session cookies which are provided on first page visit automatically?
For example cloudflare protected sites and a lot of other WAFs will set cookies on first visit otherwise requests get blocked without the correct cookies (which will result in massive 403 and not finding vulns).
As workarround, I manually visit the site and add these cookies to each nuclei request via an proxy.
Is there a way to make this possible without 3rd party tools?

Thanks and kind regards
Marc

**Comment by ehsandeep** [comment, marked as answer]
@DeC1pher check this out - https://nuclei.projectdiscovery.io/templating-guide/protocols/http/#session
